core/src/test/java/org/apache/iceberg/rest/RESTCatalogAdapter.java

```diff
@@ -285,9 +288,19 @@ public RESTClient withAuthSession(AuthSession session) {
     return this;
   }
 
+  public RESTClient withETagProvider(ETagProvider eTagProv) {
```

Comment:
I'm not sure we need this complexity right now of making this fully pluggable. I think having a separate util class that calculates the value is ok though